backend/app/lib/job_runners/ns2_remover_runner.rb

```diff
@@ -0,0 +1,65 @@
+include JSONModel
+
+class NS2RemoverRunner < JobRunner
+  register_for_job_type('ns2_remover_job',
+                        :create_permissions => :administer_system,
+                        :cancel_permissions => :administer_system)
+
+  def run
+    begin
+      modified_records = []
+
+      job_data = @json.job
+
+      RequestContext.open(:repo_id => @job.repo_id) do
+        count = 0
+        Note.each do |n|
+          parent = NotePersistentId.where(:note_id => n[:id]).first
```

Comment:
@lorawoodford This and the line below can cause an error when `parent` is `nil` (rows in the note table with no reference in the `note_persistent_id` table.)

Reply:
@lorawoodford I think the fact that the `note` table can have entries without persistent ids may be an issue itself, looking into it...